src/frontend/src/flows/register/captcha.ts

```diff
@@ -39,108 +39,101 @@ export const promptCaptchaTemplate = <T>({
 }) => {
   const copy = i18n.i18n(copyJson);
 
+  const spinnerImg: TemplateResult = html`
+    <div class="c-captcha-placeholder c-spinner-wrapper">
+      <div class="c-spinner">${spinner}</div>
+    </div>
+  `;
+  const captchaImg = (base64: string): TemplateResult =>
+    html`<div class="c-captcha-placeholder">
+      <img
+        src="data:image/png;base64,${base64}"
+        id="captchaImg"
+        class="c-image"
+        alt="captcha image"
+      />
+    </div>`;
+
+  // The various states the component can inhabit
+  type State =
+    | { status: "requesting" }
+    | { status: "prompting"; challenge: Challenge }
+    | { status: "verifying" }
+    | { status: "bad" };
+
   // We define a few Chans that are used to update the page in a
-  // reactive way; see template returned by this function
+  // reactive way based on state; see template returned by this function
+  const state = new Chan<State>({ status: "requesting" });
 
   // The image shown
-  const img = new Chan<TemplateResult>();
+  const img: Chan<TemplateResult> = state.map({
+    f: (state) =>
+      state.status === "requesting"
+        ? spinnerImg
+        : state.status === "prompting"
+        ? captchaImg(state.challenge.png_base64)
+        : Chan.unchanged,
```

Comment:
@peterpeterparker in the first review of the captcha code you pointed out that we may sometimes forget to set the value for some components (for instance in the case of the img, although in that case we didn't update the value on purpose). "Not updating the value" in now done explicitly through `Chan.unchanged`.